server/services/offendersService.js

```diff
@@ -144,19 +135,31 @@ module.exports = function createOffendersService(nomisClientBuilder, allocationC
       }
 
       const combined = [...uncategorisedResult, ...dbManualInProgress]
-      const [sentenceMap, offenceMap, pomMap, securityReferredOffenders] = await Promise.all([
-        getSentenceMap(combined, nomisClient),
-        getOffenceMap(combined, nomisClient),
+
+      const bookingIds = combined
+        .filter(o => !o.dbRecord || !o.dbRecord.catType || o.dbRecord.catType === CatType.INITIAL.name)
+        .map(o => o.bookingId)
+
+      const [prisoners, pomMap, securityReferredOffenders] = await Promise.all([
+        prisonerSearchClient.getPrisonersByBookingIds(bookingIds),
         getPomMap(combined, allocationClient),
         formService.getSecurityReferrals(agencyId, transactionalDbClient),
       ])
 
+      const prisonerMap = new Map(prisoners.map(prisoner => [prisoner.bookingId, prisoner]))
+
+      const sentenceMap = new Map(
+        prisoners
+          .filter(s => s.sentenceStartDate) // the endpoint returns records for offenders without sentences
+          .map(s => [s.bookingId, { sentenceDate: s.sentenceStartDate }])
+      )
+
       const filterIS91s = o => {
-        const offence = offenceMap.get(o.bookingId)
+        const offence = prisonerMap.get(o.bookingId)
         if (!offence) {
           return true
         }
-        if (offence.offenceCode === 'IA99000-001N' && offence.statuteCode === 'ZZ') {
+        if (offence.mostSeriousOffence === 'ILLEGAL IMMIGRANT/DETAINEE') {
```

Comment:
Description is in ES, not the codes

Reply:
There is a small risk, but there has always been a risk that a different code could be used too. Actually it is worth QA testing this case: of one of the prisoners being an IS91

Reply:
@steverendell While checking on T3 Nomis, we have found that the Offence Code and Status Code is still valid.Is it worthwhile if we keep using the same instead of description.
<img width="1232" alt="Screenshot 2022-12-16 at 14 33 23" src="https://user-images.githubusercontent.com/69511231/208121484-ed480a74-de8b-446d-9179-a90bb58dc801.png">

Reply:
Hi @nnehabharti , unfortunately I had to change this because the codes are not available in the data from the new service, only the description is present.